splunklogger.js

```diff
@@ -451,17 +629,20 @@ SplunkLogger.prototype._sendEvents = function(context, callback) {
  * @throws Will throw an error if the <code>context</code> parameter is malformed.
  * @public
  */
-SplunkLogger.prototype.send = function (context, callback) {
+SplunkLogger.prototype.send = function(context, callback) {
     callback = callback || function(){};
     context = this._initializeContext(context);
     
+    // Store the context, and its estimated length
     this.contextQueue.push(context);
+    this.eventSizes.push(Buffer.byteLength(JSON.stringify(this._makeBody(context)), "utf8"));
 
-    if (context.config.autoFlush) {
-        this.flush(callback);
-    }
-    else {
-        callback();
+    var batchOverSize = this.calculateBatchSize() > context.config.maxBatchSize;
+    var batchOverCount = this.contextQueue.length >= context.config.maxBatchCount && context.config.maxBatchCount > 0;
+
+    // Only flush if not using manual batching, and if the contextQueue is too large or has many events
+    if (context.config.autoFlush && !this._timerID && (batchOverSize || batchOverCount)) {
```

Comment:
Does this mean that `autoFlush` overrides any batch parameters (i.e. if autoFlush is false, any batch settings are ignored)?